### ISSUE TITLE
Channels view not showing latest videos for some channels since 17.88

### ISSUE THREAD

**Issue by Muffin1611** [yuliskov/SmartTube#2497]
Since updating to 17.88, if I go to "Channels", for some channels several of the newest videos aren't displayed. As an example, one channel with the issue shows me a video from 9 days ago at the top, using the context menu to "Open channel" I can see that there are four newer videos which also show in my subscriptions tab.

This is on FireTV Stick.

**Comment by ndmaque** [comment]
Similar  issue on chromecast 2021 but they don't show up in "Open Channel" but i can see them in my browser and also the Youtube TV app (on my chromecast) and my laptop browser. Quite a lot has disappeared but was previously showing. 
Example: i was going to watch "Kevin McAleer" latest vid from Sunday night but it was no longer listed along with many other of his and other channel content, i checked against both my chromecast YT app and my laptop browser and it's not just that channel. Some/most others are OK as far as i can tell.

**Comment by bibliopolist** [comment]
I experience this problem only with videos that are to be found in the "Live" section on the Youtube website (i. e., replays of live streams). For example the channel GrowingUpInScientology has most of its videos as "live", and now hardly anything is showing anymore in the app since the last update to 17.88. It worked perfectly fine before.

**Comment by DcR-NL** [comment]
Are you using a brand account? I do, and the same thing happened to me. Switching to a regular account fixes it for me. I tried downgrading to 17.62, but the same issue persisted with my brand account.

**Comment by Muffin1611** [comment]
Nope, I'm using a regular account.

**Comment by Reverend-K** [comment]
I'm having the same issue.  After the most recent update, the videos listed in each channel were weeks, sometimes months or years old.  I changed the settings to "most recent" and it fixed the issue of displaying old videos per channel, but now the channels are not listed in alphabetical order and instead listed according to which ones have the most recent videos.  Setting it to alphabetical lists the channels in the right order but shows old videos.  Sorry, I can't really describe the problem very well.

I'm also using the Fire TV Stick.

**Comment by blackwind** [comment]
Indeed, the Channels section no longer displays videos from the "Live" tab. Most likely related to ceff90d7c95af757c340e34c6f19b35a7244be19, @yuliskov. App is currently unusable for me.

**Comment by yuliskov** [comment]
It's a new algorithm. The old one is hidden behind one of channel sorting. Settings/UI/Channel section sorting.

**Comment by Muffin1611** [comment]
> It's a new algorithm. The old one is hidden behind one of channel sorting. Settings/UI/Channel section sorting.

The videos indeed show up correctly for a channel if I choose "new content" as channel sorting, so that's a workaround I'll use for now, but the default one not showing all videos for a channel isn't intended behaviour, right?

**Comment by yuliskov** [comment]
Correct. The first two options, marked as fast, are the new ones. Others are the same as the previous.
